packages/node-cli/src/services/AuthenticationService.js

```diff
@@ -0,0 +1,82 @@
+/*
+ ** Copyright (c) 2020 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+'use strict';
+
+const FileUtils = require('../utils/FileUtils');
+const NodeTranslationService = require('../services/NodeTranslationService');
+const { ERRORS, COMMAND_SETUPACCOUNT } = require('../services/TranslationKeys');
+const { FILES } = require('../ApplicationConstants');
+const { ActionResult } = require('../commands/actionresult/ActionResult');
+const { executeWithSpinner } = require('../ui/CliSpinner');
+const path = require('path');
+const SdkExecutionContext = require('../SdkExecutionContext');
+const SdkOperationResultUtils = require('../utils/SdkOperationResultUtils');
+const SdkExecutor = require('../SdkExecutor').SdkExecutor;
+
+const DEFAULT_AUTH_ID_PROPERTY = 'defaultAuthId';
+
+const COMMANDS = {
+	MANAGEAUTH: 'manageauth',
+};
+
+const FLAGS = {
+	LIST: 'list',
+};
+
+module.exports = {
+
+	setDefaultAuthentication(projectFolder, authId) {
+		try {
+			// nest the values into a DEFAULT_AUTH_ID_PROPERTY property
+			const projectConfiguration = {
+				[DEFAULT_AUTH_ID_PROPERTY]: authId,
+			};
+			FileUtils.create(path.join(projectFolder, FILES.PROJECT_JSON), projectConfiguration);
+		} catch (error) {
+			const errorMessage = error != null && error.message ? NodeTranslationService.getMessage(ERRORS.ADD_ERROR_LINE, error.message) : '';
+			throw NodeTranslationService.getMessage(ERRORS.WRITING_PROJECT_JSON, errorMessage);
+		}
+	},
+
+	getProjectDefaultAuthId(projectFolder) {
+
+		const projectFilePath = path.join(projectFolder, FILES.PROJECT_JSON);
+		if (FileUtils.exists(projectFilePath)) {
+			try {
+				const fileContentJson = FileUtils.readAsJson(projectFilePath);
+				if (!fileContentJson.hasOwnProperty(DEFAULT_AUTH_ID_PROPERTY)) {
+					throw NodeTranslationService.getMessage(ERRORS.MISSING_DEFAULT_AUTH_ID, DEFAULT_AUTH_ID_PROPERTY);
+				}
+				return fileContentJson[DEFAULT_AUTH_ID_PROPERTY];
+			} catch (error) {
+				throw NodeTranslationService.getMessage(ERRORS.WRONG_JSON_FILE, projectFilePath, error);
+			}
+		}
+	},
+
+	async getAuthIds(sdkPath) {
+
+		const sdkExecutor = new SdkExecutor(sdkPath);
```

Comment:
So, my question here is... we do not pass the sdkExecutor because in VsCode we cannot define it like in Node.js?

Maybe it is interesting to create another method that is

`async getAuthIds(sdkExecutor) `
And then getAuthIds(sdkPath) will only be used by VSCode and would call the previous one

`async getAuthIdsWithSDKPath(sdkPath) {
 return getAuthIds(new SdkExecutor(sdkPath);
}`

Names are not good, but just as example...